python_modules/dagster/dagster_tests/daemon_sensor_tests/test_sensor_run.py

```diff
@@ -1750,6 +1757,61 @@ def test_launch_once(caplog, executor, instance, workspace_context, remote_repo)
         )
 
 
+def test_duplicate_run_key_within_tick_launches_once(
+    executor: ThreadPoolExecutor,
+    instance: DagsterInstance,
+    workspace_context: WorkspaceProcessContext,
+    remote_repo: RemoteRepository,
+):
+    freeze_datetime = create_datetime(
+        year=2019,
+        month=2,
+        day=27,
+        hour=23,
+        minute=59,
+        second=59,
+    )
+    with (
+        freeze_time(freeze_datetime),
+        patch.object(DagsterInstance, "get_ticks", wraps=instance.get_ticks) as mock_get_ticks,
+    ):
+        sensor = remote_repo.get_sensor("dup_run_key_sensor")
+        instance.add_instigator_state(
+            InstigatorState(
+                sensor.get_remote_origin(),
+                InstigatorType.SENSOR,
+                InstigatorStatus.RUNNING,
+            )
+        )
+        assert instance.get_runs_count() == 0
+
+        assert mock_get_ticks.call_count == 0
+        ticks = instance.get_ticks(sensor.get_remote_origin_id(), sensor.selector_id)
+        assert mock_get_ticks.call_count == 1
+        assert len(ticks) == 0
+
+        evaluate_sensors(workspace_context, executor)
+
+        # call another time to get previous tick
+        assert mock_get_ticks.call_count == 2
+
+        wait_for_all_runs_to_start(instance)
+
+        assert instance.get_runs_count() == 1
+        run = instance.get_runs()[0]
```

Comment:
could add something that asserts that it's the first of the two always (e.g. putting a tag on that run request and verifying that that run is the one that was fired